lib/node_modules/@stdlib/array/log/test/test.js

```diff
@@ -0,0 +1,457 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var proxyquire = require( 'proxyquire' );
+var tape = require( 'tape' );
+var log = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof log, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if not provided a string for the first argument (unary)', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		-3,
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided '+values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			log( value );
+		};
+	}
+});
+
+tape( 'the function throws an error if not provided a string for the first argument (n-ary)', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		-3,
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), TypeError, 'throws an error when provided '+values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			log( value, [ 1, 2, 3 ] );
+		};
+	}
+});
+
+tape( 'the function throws an error if not provided collections with the same length', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		[ 1, 2, 3 ],
+		[ 4, 5 ],
+		[ 6 ],
+		[]
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[ i ] ), RangeError, 'throws an error when provided '+values[ i ] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			log( '%d, %d', value, [ 1, 2, 3, 4 ] );
+		};
+	}
+});
+
+tape( 'the function prints the correct result when provided 1 array', function test( t ) {
+	var expected;
+	var values;
+	var log;
+	var i;
+	var j;
+
+	log = proxyquire( './../lib/main.js', {
+		'./logger.js': logger
+	});
+
+	j = 0;
+
+	values = [
+		[ 1, 2, 3 ],
+		[ 4, 5 ],
+		[ 6 ],
+		[ 'beep' ]
+	];
+
+	expected = [ '1', '2', '3', '4', '5', '6', 'beep' ];
+
+	for ( i = 0; i < values.length; i++ ) {
+		log( '%s', values[ i ] );
+	}
+	t.end();
+
+	function logger( str ) {
+		t.equal( str, expected[ j ], 'returns expected value' );
+		j += 1;
+	}
+});
+
+tape( 'the function prints the correct result when provided 2 arrays with the same length', function test( t ) {
```

Comment:
```suggestion
tape( 'the function prints the correct result when provided two arrays', function test( t ) {
```

Given the assertion test above, it's assumed that provided arrays must be the same length.